src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -0,0 +1,291 @@
+            std::string username = std::get<0> (users[i]);
+            std::string user_shares = std::get<1> (users[i]);
+            std::string user_usage = std::get<2> (users[i]);
+            user_node = std::make_shared<weighted_tree_node_t> (
+                            parent_bank,
+                            username,
+                            true,
+                            std::stoll (user_shares),
+                            std::stoll (user_usage));
+            node->add_child (user_node);
+            leaf_bank_usage += std::stoi (user_usage);
+        }
+        // aggregate usage from user nodes up to their respective banks
+        // and up to the root bank
+        for (std::shared_ptr<weighted_tree_node_t> curr_ancestor = node;
+             curr_ancestor != nullptr;
+             curr_ancestor = curr_ancestor->get_parent ()) {
+
+            curr_ancestor->set_usage (curr_ancestor->get_usage ()
+                                      + leaf_bank_usage);
+        }
+    }
+    // otherwise, this bank has sub banks, so call this
+    // function again with the first sub bank it found
+    else {
+        parent_bank = node;
+        while (rc == SQLITE_ROW) {
+            // execute SQL statement
+            std::string bank = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_sub_banks_stmt, 0));
+            banks.push_back (bank);
+            rc = sqlite3_step (b_select_sub_banks_stmt);
+        }
+        for (std::string b : banks) {
+            // reset the prepared statements back to their initial state and
+            // clear their bindings
+            sqlite3_clear_bindings (b_select_associations_stmt);
+            sqlite3_reset (b_select_associations_stmt);
+            sqlite3_clear_bindings (b_select_sub_banks_stmt);
+            sqlite3_reset (b_select_sub_banks_stmt);
+            sqlite3_clear_bindings (b_select_shares_stmt);
+            sqlite3_reset (b_select_shares_stmt);
+            get_sub_banks (DB,
+                           b,
+                           parent_bank,
+                           b_select_shares_stmt,
+                           b_select_sub_banks_stmt,
+                           b_select_associations_stmt);
+        }
+    }
+    return root_ptr;
+}
+
+std::vector<std::shared_ptr<weighted_tree_node_t>>
+load_accounting_db (const std::string &path) {
+    // SQL statements to retrieve data from flux-accounting database
+    sqlite3_stmt *b_select_root_bank_stmt;
+    sqlite3_stmt *b_select_shares_stmt;
+    sqlite3_stmt *b_select_sub_banks_stmt;
+    sqlite3_stmt *b_select_associations_stmt;
+
+    sqlite3 *DB;
+    int rc = 0;
+
+    // open FluxAccounting DB in read-write mode; if it does not exist yet,
+    // create a new database file
+    rc = sqlite3_open_v2 (path.c_str (), &DB, SQLITE_OPEN_READWRITE, NULL);
+    if (rc) {
+        std::cerr << "error opening DB: " << sqlite3_errmsg (DB) << std::endl;
+        exit (EXIT_FAILURE);
```

Comment:
Since this transitioned to being a library as opposed to a front-end utility, I don't think exiting is the right move.  I'd return a valid error value, like a null pointer.